config/initializers/active_model_serializers.rb

```diff
@@ -0,0 +1,166 @@
+    class Attributes < Base
+
+      def serializable_hash(options = nil)
+        return nil if serializer.nil? || serializer.object.nil?
+
+        is_collection = serializer.respond_to?(:each)
+        serializers = is_collection ? serializer : [serializer]
+        data = resource_objects_for(serializers)
+
+        result = is_collection ? data : data[0]
+
+        self.class.transform_key_casing!(result, instance_options)
+      end
+
+    private
+
+      def resource_objects_for(serializers)
+        serializers.map { |serializer| resource_object_for(serializer) }
+      end
+
+    end
+
+    class AttributesWithIncluded < Base
+
+      def serializable_hash(options = nil)
+        return {} if serializer.object.nil?
+
+        is_collection = serializer.respond_to?(:each)
+        serializers = is_collection ? serializer : [serializer]
+        primary_data, included = resource_objects_for(serializers)
+
+        hash = {}
+        hash[root] = is_collection ? primary_data : primary_data[0]
+        hash[:included] = included if included.any?
+        hash[meta_key] = meta unless meta.blank?
+
+        self.class.transform_key_casing!(hash, instance_options)
+      end
+
+    private
+
+      def resource_objects_for(serializers)
+        @primary = []
+        @included = {}
+        @resource_identifiers = Set.new
+        serializers.each { |serializer| process_resource(serializer) }
+        serializers.each { |serializer| process_relationships(serializer) }
+
+        [@primary, @included]
+      end
+
+      def process_resource(serializer)
+        return false unless resource_already_processed?(serializer)
+
+        @primary << resource_object_for(serializer)
+        true
+      end
+
+      def process_relationship_resource(serializer, json_key)
+        return false unless resource_already_processed?(serializer)
+
+        @included[json_key] << resource_object_for(serializer)
+        true
+      end
+
+      def process_relationships(serializer)
+        serializer.associations.each do |association|
+          process_relationship(association.serializer, association.key)
+        end
+      end
+
+      def process_relationship(serializer, key)
+        json_key = key || serializer.json_key.to_s.pluralize
+        @included[json_key] ||= []
+
+        if serializer.respond_to?(:each)
+          serializer.each { |s| process_relationship(s, key) }
+          return
+        end
+        return unless serializer && serializer.object
+        return unless process_relationship_resource(serializer, json_key)
+
+        process_relationships(serializer)
+      end
+
+      def resource_already_processed?(serializer)
+        resource_identifier = ResourceIdentifier.new(serializer, instance_options).as_json
+        @resource_identifiers.add?(resource_identifier)
+      end
+
+      def meta
+        instance_options.fetch(:meta, nil)
+      end
+
+      def meta_key
+        instance_options.fetch(:meta_key, 'meta'.freeze)
```

Comment:
Style/StringLiterals: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.